Blizzard.IO.RabbitMQ/IRabbitPublisher.cs

```diff
@@ -0,0 +1,10 @@
+﻿using Blizzard.IO.RabbitMQ.Entities;
+
+namespace Blizzard.IO.RabbitMQ
+{
+    public interface IRabbitPublisher<TData>
```

Comment:
Add the variant `in` notation on `TData`

Reply:
Why do we need the variant in here?

Reply:
in order to make the `TData` immutable inside the class

Reply:
you need to explain it to me personally